File: backend/app/services/labels.py
"""Canonical label normalization for API and client contracts."""

from typing import Optional

SAFE_LABELS = frozenset({"safe", "benign", "ham", "legitimate", "legit"})
PHISHING_LABELS = frozenset({"phishing", "spam", "malicious", "fraud"})


def normalize_label(label: Optional[str]) -> str:
    if not label:
        return "unknown"
    normalized = str(label).strip().lower()
    if normalized in SAFE_LABELS:
        return "safe"
    if normalized in PHISHING_LABELS:
        return "phishing"
    if normalized == "unknown":
        return "unknown"
    return normalized


def is_threat_label(label: Optional[str]) -> bool:
    return normalize_label(label) == "phishing"


def is_safe_label(label: Optional[str]) -> bool:
    return normalize_label(label) == "safe"

File: frontend/src/utils/labels.js
const SAFE = new Set(["safe", "benign", "ham", "legitimate"]);
const PHISHING = new Set(["phishing", "spam", "malicious"]);

export const normalizeLabel = (label) => {
    const n = (label || "unknown").toString().toLowerCase();
    if (SAFE.has(n)) return "safe";
    if (PHISHING.has(n)) return "phishing";
    return n;
};

export const isThreatLabel = (label) => normalizeLabel(label) === "phishing";
